plugins/modules/aci_l3out.py

```diff
@@ -394,26 +408,20 @@ def main():
 
     if l3protocol:
         if "eigrp" in l3protocol and asn is None:
-            module.fail_json(
-                msg="Parameter 'asn' is required when l3protocol is 'eigrp'"
-            )
+            module.fail_json(msg="Parameter 'asn' is required when l3protocol is 'eigrp'")
         if "eigrp" not in l3protocol and asn is not None:
-            module.warn(
-                "Parameter 'asn' is only applicable when l3protocol is 'eigrp'. The ASN will be ignored"
-            )
+            module.warn("Parameter 'asn' is only applicable when l3protocol is 'eigrp'. The ASN will be ignored")
+        if "ospf" in l3protocol and ospf is None:
+            module.fail_json(msg="ospf specifications are required when l3protocol is 'ospf'")
```

Comment:
Is this really the case, since apic seems to default to parameters if I look at the model-doc. Could you also cross check what is required on the UI?